packages/gestalt/src/AlertModal.js

```diff
@@ -0,0 +1,202 @@
+   */
+  type?: 'default' | 'warning' | 'error',
+  /**
+   * Main action for users to take on AlertModal. If `href` is supplied, the action will serve as a link. See [OnLinkNavigationProvider](https://gestalt.pinterest.systems/web/utilities/onlinknavigationprovider) to learn more about link navigation.
+   * If no `href` is supplied, the action will be a button.
+   * The `accessibilityLabel` should follow the [Accessibility guidelines](https://gestalt.pinterest.systems/web/alertmodal#Accessibility).
+   */
+  primaryAction: {|
+    accessibilityLabel: string,
+    disabled?: boolean,
+    href?: string,
+    label: string,
+    onClick?: AbstractEventHandler<
+      | SyntheticMouseEvent<HTMLButtonElement>
+      | SyntheticMouseEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLButtonElement>,
+      {| dangerouslyDisableOnNavigation: () => void |},
+    >,
+    rel?: 'none' | 'nofollow',
+    target?: null | 'self' | 'blank',
+  |},
+  /**
+   * Secondary action for users to take on AlertModal. If `href` is supplied, the action will serve as a link. See [OnLinkNavigationProvider](https://gestalt.pinterest.systems/web/utilities/onlinknavigationprovider) to learn more about link navigation.
+   * If no `href` is supplied, the action will be a button.
+   * The `accessibilityLabel` should follow the [Accessibility guidelines](https://gestalt.pinterest.systems/web/alertmodal#Accessibility).
+   */
+  secondaryAction?: {|
+    accessibilityLabel: string,
+    disabled?: boolean,
+    href?: string,
+    label: string,
+    onClick?: AbstractEventHandler<
+      | SyntheticMouseEvent<HTMLButtonElement>
+      | SyntheticMouseEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLButtonElement>,
+      {| dangerouslyDisableOnNavigation: () => void |},
+    >,
+    rel?: 'none' | 'nofollow',
+    target?: null | 'self' | 'blank',
+  |},
+|};
+
+const ICON_COLOR_MAP = {
+  error: {
+    icon: 'workflow-status-problem',
+    color: 'error',
+  },
+  warning: {
+    icon: 'workflow-status-warning',
+    color: 'warning',
+  },
+};
+
+function Header({
+  type,
+  heading,
+  onDismiss,
+}: {|
+  type: 'default' | 'warning' | 'error',
+  heading: string,
+  onDismiss: () => void,
+|}) {
+  return (
+    <Box padding={6}>
+      <Flex flex="grow" alignItems="center">
+        {type !== 'default' && (
+          <Box marginEnd={4}>
+            <Icon
+              size="20"
+              accessibilityLabel={type}
+              icon={ICON_COLOR_MAP[type].icon}
+              color={ICON_COLOR_MAP[type].color}
+            />
+          </Box>
+        )}
+        <Flex.Item flex="grow">
+          {' '}
+          <Heading size="400" accessibilityLevel={1}>
+            {heading}
+          </Heading>
+        </Flex.Item>
+        {type === 'default' && (
+          <Box marginStart={6}>
```

Comment:
It seems like the parent Flex could have a `gap={4}`, with this Box setting `marginStart={2}`? That would obviate the need for the other `marginEnd={4}`